substra/sdk/client.py

```diff
@@ -78,19 +221,79 @@ class Client:
 
     def __init__(
         self,
+        *,
+        client_name: Optional[str] = None,
+        configuration_file: Optional[pathlib.Path] = None,
         url: Optional[str] = None,
         token: Optional[str] = None,
-        retry_timeout: int = DEFAULT_RETRY_TIMEOUT,
-        insecure: bool = False,
-        backend_type: schemas.BackendType = schemas.BackendType.REMOTE,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        retry_timeout: Optional[int] = None,
+        insecure: Optional[bool] = None,
+        backend_type: Optional[schemas.BackendType] = None,
     ):
-        self._retry_timeout = retry_timeout
-        self._token = token
+        if configuration_file and not client_name:
+            raise exceptions.BadConfiguration("Configuration file cannot be used because no `client_name` was given.")
```

Comment:
Not a fan of this error name. Custom errors should at least end by `*Error`. 
`ConfigurationInfoError` or `ClientNameError` would be better IMO.